File: fix_typos.js

const fs = require('fs');

const filePath = "Mühazirə_ERQONOMİKA.md";

const replacements = [
    { pattern: /Frqonomik/g, replacement: "Erqonomik" },
    { pattern: /Frqonomiya/g, replacement: "Erqonomiya" },
    { pattern: /antropoloyi/g, replacement: "antropoloji" },
    { pattern: /psixoloyi/g, replacement: "psixoloji" },
    { pattern: /fizioloyi/g, replacement: "fizioloji" },
    { pattern: /bioloyi/g, replacement: "bioloji" },
    { pattern: /texnoloyi/g, replacement: "texnoloji" },
    { pattern: /metodoloyi/g, replacement: "metodoloji" },
    { pattern: /gigenik/g, replacement: "gigienik" },
    { pattern: /nəzə almaqla/g, replacement: "nəzərə almaqla" },
    { pattern: /sosial-psixoloyi/g, replacement: "sosial-psixoloji" },
    // Regex for 'oloyi' ending word boundaries
    { pattern: /([a-z])oloyi\b/g, replacement: "$1oloji" } 
];

try {
    let content = fs.readFileSync(filePath, 'utf8');
    console.log(`Read ${content.length} characters.`);

    let count = 0;
    replacements.forEach(item => {
        // Count matches
        const matches = content.match(item.pattern);
        if (matches) {
            console.log(`Replaced '${item.pattern}': ${matches.length} times.`);
            content = content.replace(item.pattern, item.replacement);
            count += matches.length;
        }
    });

    fs.writeFileSync(filePath, content, 'utf8');
    console.log(`Done. Total replacements: ${count}`);

} catch (e) {
    console.error(`Error: ${e}`);
}

File: fix_typos.py

import re

file_path = "Mühazirə_ERQONOMİKA.md"

replacements = [
    (r"Frqonomik", "Erqonomik"),
    (r"Frqonomiya", "Erqonomiya"),
    (r"antropoloyi", "antropoloji"),
    (r"psixoloyi", "psixoloji"),
    (r"fizioloyi", "fizioloji"),
    (r"bioloyi", "bioloji"),
    (r"texnoloyi", "texnoloji"),
    (r"metodoloyi", "metodoloji"),
    (r"gigenik", "gigienik"),
    (r"nəzə almaqla", "nəzərə almaqla"),
    (r"sosial-psixoloyi", "sosial-psixoloji"),
    # General fix for 'loyi' ending which is usually 'loji' in these contexts
    (r"([a-z])oloyi\b", r"\1oloji"), 
]

try:
    with open(file_path, "r", encoding="utf-8") as f:
        content = f.read()

    original_len = len(content)
    print(f"Read {original_len} characters.")

    count = 0
    for pattern, replacement in replacements:
        new_content, n = re.subn(pattern, replacement, content)
        if n > 0:
            print(f"Replaced '{pattern}' with '{replacement}': {n} times.")
            content = new_content
            count += n

    with open(file_path, "w", encoding="utf-8") as f:
        f.write(content)

    print(f"Done. Total replacements: {count}")

except Exception as e:
    print(f"Error: {e}")
